src/main/java/com/importer/importer/service/StudentService.java

```diff
@@ -0,0 +1,88 @@
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+public class StudentService {
+
+    @Autowired
+    private LogRepository logRepository;
+    @Transactional
+    public String postAllStudents(List<StudentCreationDto> studentCreationDtos){
+        StringBuilder responseBuilder = new StringBuilder();
+        HttpHeaders headers = new HttpHeaders();
+        Jwt jwt = (Jwt) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        String jwtToken = jwt.getTokenValue();
+        headers.add("Authorization", "Bearer "+ jwtToken );
+        headers.setAccept(List.of(MediaType.APPLICATION_JSON));
+        for(StudentCreationDto studentCreationDto : studentCreationDtos){
+
+            HttpEntity<StudentCreationDto> entity = new HttpEntity<StudentCreationDto>(studentCreationDto, headers);
+
+            String status = "";
+            String responseBody = "";
+            int statusCode = 0;
+            String responseMessage = "";
+            try {
+                ResponseEntity<String> responseEntity = new RestTemplate().exchange("http://localhost:8086/students", HttpMethod.POST, entity, String.class);
+                responseBuilder.append("Status Code: ").append(responseEntity.getStatusCode()).append(System.lineSeparator());
+                responseBuilder.append("Response Body: ").append(responseEntity.getBody()).append(System.lineSeparator());
+                status = responseEntity.getStatusCode().toString();
+                statusCode = Integer.parseInt(status.substring(0, status.indexOf(" ")));
+                responseMessage = status.substring(status.indexOf(" ")+1);
+                responseBody = "Student Created Successfully";
+
+            } catch (HttpClientErrorException.BadRequest ex) {
+
+                status = ex.getStatusCode().toString();
+                statusCode = ex.getStatusCode().value();
+                responseMessage = status.substring(status.indexOf(" ")+1);
+                if(!ex.getResponseBodyAsString().isEmpty())
+                    responseBody = ex.getResponseBodyAsString();
+            } catch (HttpClientErrorException ex) {
+                // Handle other 4xx errors
+                status = ex.getStatusCode().toString();
+                statusCode = ex.getStatusCode().value();
+                responseMessage = status.substring(status.indexOf(" ")+1);
+                if(!ex.getResponseBodyAsString().isEmpty())
+                    responseBody = ex.getResponseBodyAsString();
+
+            } catch (Exception ex) {
+                responseBuilder.append("Error: ").append(ex.getMessage()).append(System.lineSeparator());
+                statusCode = 500;
+                responseMessage = "Internal Server Error";
+            }
+            finally {
```

Comment:
Instead of using the finally block, we can write this code in the try block itself.

Reply:
removed, I added because to avoid repetitive code